contrib/mesos/pkg/executor/executor.go

```diff
@@ -208,23 +210,74 @@ func (k *KubernetesExecutor) isDone() bool {
 	}
 }
 
+func (k *KubernetesExecutor) NodeInfo() (ni NodeInfo) {
+	k.nodeInfoLock.Lock()
+	defer k.nodeInfoLock.Unlock()
+
+	// clone k.nodeInfo
```

Comment:
Does that work with custom structs? I thoughts it uses pregenerated deep copy functions.